### PR TITLE
Feature/search filter sales

### PR DESCRIPTION
**Pull request by Snowflake8**
Trying to make filters work first with filters and then plug that into firebase
## Pull reviewers stats
Stats for the last 180 days:
|                                                                                                                                                               | User            | Total reviews | Median time to review                                                                                                                                                                                                                                                                                                                                                                                                                             | Total comments |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------- | ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------- |
| <a href=https://github.com/Huntsekkern><img src="https://avatars.githubusercontent.com/u/44240415?v=4" width="20"></a>                                        | Huntsekkern     | 9             | [1h 16m](https://app.flowwer.dev/charts/review-time/~(u~(i~'44240415~n~'Huntsekkern)~p~180~r~(~(d~'qpumtn~t~'1745)~(d~'qpt0zy~t~'2wmx)~(d~'qpt88n~t~'33vm)~(d~'qptd1d~t~'29r)~(d~'qptd3y~t~'2cc)~(d~'qpt4tp~t~'2b1)~(d~'qpuo7b~t~'mk)~(d~'qq6uip~t~'3j5)~(d~'qq8faa~t~'qv)~(d~'qq8fiz~t~'bv)~(d~'qqjij5~t~'i6m)~(d~'qqkomy~t~'z8i)~(d~'qqkozj~t~'zl3)~(d~'qqkp4h~t~'zq1)~(d~'qqvo8r~t~'14d))))                                                    | **10**         |
| <a href=https://github.com/Snowflake8><img src="https://avatars.githubusercontent.com/u/44499990?v=4" width="20"></a>                                         | Snowflake8      | 10            | [6h 21m](https://app.flowwer.dev/charts/review-time/~(u~(i~'44499990~n~'Snowflake8)~p~180~r~(~(d~'qpt3s9~t~'2zf8)~(d~'qpx29n~t~'1vgi)~(d~'qq0q7g~t~'5jeb)~(d~'qq6k9y~t~'8hk)~(d~'qq6kh9~t~'8ov)~(d~'qq6r5v~t~'6b)~(d~'qqdm38~t~'1fge)~(d~'qq6ncv~t~'8609)~(d~'qq8el9~t~'y2h)~(d~'qqjh1g~t~'gox)~(d~'qqjh87~t~'7lg)~(d~'qqjlvf~t~'b)~(d~'qqxzb3~t~'ikt)~(d~'qqsnpq~t~'qn))))                                                                       | 9              |
| <a href=https://github.com/RaulinN><img src="https://avatars.githubusercontent.com/u/44499915?u=c029696ff2de0902f905fbf35188359e94e08d88&v=4" width="20"></a> | RaulinN         | 10            | [3h 28m](https://app.flowwer.dev/charts/review-time/~(u~(i~'44499915~n~'RaulinN)~p~180~r~(~(d~'qptcqf~t~'1yt)~(d~'qpq5rb~t~'5xjz)~(d~'qq73ig~t~'jy6)~(d~'qq7vmj~t~'1t5)~(d~'qq7qbn~t~'1bz)~(d~'qpuwak~t~'8pt)~(d~'qq73am~t~'28)~(d~'qqhz4w~t~'87bj)~(d~'qq73x3~t~'8mkh)~(d~'qq821x~t~'lj5)~(d~'qqr6wy~t~'aju)~(d~'qreyev~t~'f))))                                                                                                                 | 7              |
| <a href=https://github.com/so7fie><img src="https://avatars.githubusercontent.com/u/10992808?v=4" width="20"></a>                                             | so7fie          | 9             | [1d 13h 27m](https://app.flowwer.dev/charts/review-time/~(u~(i~'10992808~n~'so7fie)~p~180~r~(~(d~'qpuy0b~t~'1v)~(d~'qq0p5z~t~'5hby)~(d~'qq7qp0~t~'hf)~(d~'qq0t0v~t~'5m7q)~(d~'qq0t2c~t~'5m97)~(d~'qqew7j~t~'2pkp)~(d~'qqf92s~t~'32fy)~(d~'qqjbho~t~'1ux)~(d~'qqkh75~t~'dr0)~(d~'qqrud7~t~'7qx2)~(d~'qqy5l4~t~'5im1)~(d~'qr8kh3~t~'1kfr))))                                                                                                        | 4              |
| <a href=https://github.com/DGoedtkindtEPFL><img src="https://avatars.githubusercontent.com/u/61052800?v=4" width="20"></a>                                    | DGoedtkindtEPFL | **12**        | [19h 30m](https://app.flowwer.dev/charts/review-time/~(u~(i~'61052800~n~'DGoedtkindtEPFL)~p~180~r~(~(d~'qpuy16~t~'2q)~(d~'qpuoy1~t~'co)~(d~'qpup0b~t~'19at)~(d~'qplma9~t~'1e2x)~(d~'qpuugt~t~'am9h)~(d~'qq6s0w~t~'8gm)~(d~'qq6rfm~t~'fn8)~(d~'qqjjxz~t~'9s4m)~(d~'qq9yj2~t~'1i5c)~(d~'qqxoq1~t~'21ln)~(d~'qqjmgc~t~'dy)~(d~'qqrtyv~t~'xlr)~(d~'qqioj1~t~'2x8a)~(d~'qqionj~t~'2xcs)~(d~'qqioou~t~'2xe3)~(d~'qqipcq~t~'2y1z)~(d~'qqru2n~t~'7qmi)))) | 4              |
| <a href=https://github.com/Laniakea1999><img src="https://avatars.githubusercontent.com/u/43450242?v=4" width="20"></a>                                       | Laniakea1999    | 3             | [**8m**](https://app.flowwer.dev/charts/review-time/~(u~(i~'43450242~n~'Laniakea1999)~p~180~r~(~(d~'qpupsq~t~'d4)~(d~'qq7oa5~t~'1n)~(d~'qqeyk1~t~'2rx7))))                                                                                                                                                                                                                                                                                        | 0              |

### CODE REVIEW

**Comment by Snowflake8** [comment]
Attention ce build ne passe pas a cause de code climate (si vous cliquez sur la croix du commit) contrairement a ce qui est marqué sur mon UI

**Review by so7fie — APPROVED** [review]

File: app/src/main/java/com/github/polybooks/ListSalesActivity.kt [hunk 1]
```diff
@@ -9,17 +10,25 @@ import com.github.polybooks.core.database.SalesAdapter
...
+    companion object {
+        val EXTRA_SALE_QUERY :String = "saleQuery"
+    }
+    private val TAG: String = "ListSaleActivity"
+
     private lateinit var mRecycler : RecyclerView
     private lateinit var mAdapter : SalesAdapter
     private val mLayout : RecyclerView.LayoutManager = LinearLayoutManager(this)
-    private val initalBooks : List<Sale> = listOf(Sale("Book1", 1, 23.00f, BookCondition.GOOD, Timestamp(format.parse("2016-05-05")!!), SaleState.ACTIVE))
+    private val initalBooks : List<Sale> = listOf(Sale("Book1", 1, 23.00f, BookCondition.GOOD, Timestamp(format.parse("2016-05-05")!!.time), SaleState.ACTIVE))
```
Comment (DGoedtkindtEPFL) [inline app/src/main/java/com/github/polybooks/ListSalesActivity.kt:31]: Est-ce que c'est une bonne chose que d'avoir des livres initiaux? Est-ce que la liste ne devrait pas être vide?

File: app/src/main/java/com/github/polybooks/ListSalesActivity.kt [hunk 2]
```diff
@@ -9,17 +10,25 @@ import com.github.polybooks.core.database.SalesAdapter
...
  * Activity to list all active sales
  * @property saleQuery the query listing what is to be shown
  */
+// TODO enlever parametre
 class ListSalesActivity(private val saleQuery: SaleQuery = DummySalesQuery()) : AppCompatActivity() {
+
+    companion object {
+        val EXTRA_SALE_QUERY :String = "saleQuery"
+    }
+    private val TAG: String = "ListSaleActivity"
```
Comment (DGoedtkindtEPFL) [inline app/src/main/java/com/github/polybooks/ListSalesActivity.kt:26]: Le TAG n'est pas utilisé. Pourquoi est-ce que une constante est dans le companion Object et l'autre pas? ça pourrait aussi être en "const val" plutôt que dans le companion object.

File: app/src/main/java/com/github/polybooks/core/database/interfaces/SaleDatabase.kt [hunk 3]
```diff
@@ -40,30 +44,32 @@ interface SaleDatabase {
  * A SaleQuery is a builder for a query to the database that will yield Sales.
  * Most methods return themselves for function chaining.
  * */
-interface SaleQuery : Query<Sale> {
+interface SaleQuery : Query<Sale>, Serializable {
```
Comment (DGoedtkindtEPFL) [inline app/src/main/java/com/github/polybooks/core/database/interfaces/SaleDatabase.kt:47]: Pourquoi ?

**Review by DGoedtkindtEPFL — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/polybooks/ListSalesActivity.kt (hunk 1 above)
Comment (Snowflake8) [reply app/src/main/java/com/github/polybooks/ListSalesActivity.kt:31]: C'est temporaire, pour simuler les livres du cache

File: app/src/main/java/com/github/polybooks/ListSalesActivity.kt (hunk 2 above)
Comment (Snowflake8) [reply app/src/main/java/com/github/polybooks/ListSalesActivity.kt:26]: TAG est utilisé dans les messages de debug Log.d et j'ai juste laissé si on en a besoin plus tard. EXTRA_SALE_QUERY est necessaire pour recuperer l'argumement passé a l'activité

File: app/src/main/java/com/github/polybooks/core/database/interfaces/SaleDatabase.kt (hunk 3 above)
Comment (Snowflake8) [reply app/src/main/java/com/github/polybooks/core/database/interfaces/SaleDatabase.kt:47]: C'est nécessaire pour passer une instance de la classe d'une activité à une autre

Comment (Snowflake8) [reply app/src/main/java/com/github/polybooks/core/database/interfaces/SaleDatabase.kt:47]: On en avait parlé il y a genre 3 semaines

Comment (so7fie) [reply app/src/main/java/com/github/polybooks/core/database/interfaces/SaleDatabase.kt:47]: @DGoedtkindtEPFL https://stackoverflow.com/questions/2736389/how-to-pass-an-object-from-one-activity-to-another-on-android),

**Review by DGoedtkindtEPFL — APPROVED** [review]

**Review by DGoedtkindtEPFL — COMMENTED** [review]
Il faudrait au fait supprimer les serialialisable de main avant de merge avec main

**Pull request merged**